sdk/src/main/java/org/coralibre/android/sdk/internal/device_info/DeviceList.java

```diff
@@ -0,0 +1,81 @@
+package org.coralibre.android.sdk.internal.device_info;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import androidx.core.content.SharedPreferencesCompat;
+
+import org.coralibre.android.sdk.BuildConfig;
+import org.coralibre.android.sdk.R;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+
+import static org.coralibre.android.sdk.internal.device_info.ConfidenceLevel.NONE;
+import static org.coralibre.android.sdk.internal.device_info.ConfidenceLevel.getConfidenceLevel;
+
+public class DeviceList {
+    private static final String TAG = DeviceInfo.class.toString();
+
+    private static final String DEVICE_INFO_KEY = "DeviceList_device_info_key";
+
+    private static final int LIST_FILE_ID = R.raw.en_calibration_2020_08_12;
+    private static final int LIST_LINE_COUNT = 11810;
+    private static final DeviceInfo DEFAULT_INFO = new DeviceInfo(
+            "unknown", "unknown", "unknown", 0, 0, NONE);
+
+    final DeviceInfo[] deviceList;
+
+    public DeviceList(Context context) throws IOException {
+        deviceList = loadDeviceInfoTable(context);
+    }
+
+    public static DeviceInfo[] loadDeviceInfoTable(Context context) throws IOException {
+        BufferedReader infoCVS = new BufferedReader(new InputStreamReader(
+                context.getResources().openRawResource(LIST_FILE_ID)));
+        DeviceInfo[] deviceList = new DeviceInfo[LIST_LINE_COUNT - 1];
+        String line = infoCVS.readLine(); // discard the first line as it just contains the header
+        int i = 0;
+        while ((line = infoCVS.readLine()) != null) {
+            deviceList[i] = new DeviceInfo(line);
+            i++;
+        }
+        infoCVS.close();
+        return deviceList;
+    }
+
+    public DeviceInfo getInfo(final String manufacturer, final String device, final String model) {
+        int index = Arrays.binarySearch(deviceList, new DeviceInfo(manufacturer, device, model, 0, 0, NONE));
+        if (0 < index && deviceList[index].getModel().equals(model)) {
+            return deviceList[index];
+        } else {
+            return new DeviceInfo(manufacturer, device, model,
+                    DEFAULT_INFO.getRssi_connection(),
+                    DEFAULT_INFO.getTx(),
+                    DEFAULT_INFO.getCalibration_confidence());
+        }
+    }
+
+    public static DeviceInfo getOnwDeviceInfo(Context context) throws IOException {
```

Comment:
What does `Onw` mean? Is that supposed to be `Own`? Also, since the thrown exception is [wrapped](https://github.com/CoraLibre/CoraLibre-android-sdk/pull/46/files?diff=unified&w=1#diff-5db70856554e8145b7bb034da93f52d3R105) in a `RuntimeException`, could you have this function throw a `RuntimeException` directly?

Reply:
Own means that this function will return the DeviceInfo of the device the function is executed on.

Reply:
Yeah, but there is a typo: `Onw` instead of `Own`